src/main/java/shef/servlets/PreferenceServlet.java

```diff
@@ -0,0 +1,127 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shef.servlets;
+
+import shef.data.User;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Arrays;
+import java.util.Map;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+
+/** Servlet to set the user's preferences. */
+@WebServlet("/preferences")
+public class PreferenceServlet extends HttpServlet {
+
+  /** This is a mapping of responses on the quiz to a list of preferences.
+    * For example, if the user selected "vegetarian" on the quiz, we must
+    * add all of the preferences defined here. (MEAT-FREE, VEGETARIAN, etc.)
+    */
+  public Map<String, List<String>> preferenceMap = new HashMap<String, List<String>>() {
+	  {
+      // Question 1: What type of person are you?
+      put("student", Arrays.asList("CHEAP", "INEXPENSIVE", "EASY TO MAKE", "EASY", "SIMPLE", "QUICK", "QUICK TO MAKE"));
+      put("influencer", Arrays.asList("DESSERT", "CAKE", "ICE CREAM"));
+      put("parent", Arrays.asList("CHEAP", "INEXPENSIVE", "EASY TO MAKE", "EASY", "SIMPLE", "QUICK", "QUICK TO MAKE"));
+      put("adventurous", Arrays.asList("AFRICAN, AMERICAN, ASIAN, CARRIBEAN", "EASTERN EUROPEAN", "WESTERN EUROPEAN", "LATIN", "LATIN AMERICAN", "MEDITERRANEAN", "MIDDLE EASTERN", "SOUTH ASIAN"));
+      put("picky-eater", Arrays.asList("SIMPLE"));
+      put("sweets-lover", Arrays.asList("SWEET", "CAKE", "COOKIES", "SUGARY", "DESSERT", "PIE", "ICE CREAM"));
+      put("newbie", Arrays.asList("EASY TO MAKE", "EASY", "SIMPLE", "QUICK", "QUICK TO MAKE"));
+
+      // Question 2: What dietary requirements do you have?
+		  put("vegetarian", Arrays.asList("MEAT-FREE", "MEAT FREE", "NO MEAT", "VEGETARIAN", "VEGAN", "FISH-FREE", "FISH FREE", "NO FISH"));
```

Comment:
Quick fix: some of the indentation is off here

Reply:
I have no idea why I appears like this on github, the weird indentation isn't there for me locally

Reply:
probably difference between using space and tab. Tabs appear as 8 spaces in github by default.